### DISCUSSION TITLE
How to use mongoose with atlas search index

### DISCUSSION THREAD

**Discussion by vothvovo** [Automattic/mongoose#12779] [Q&A] [answered]
I made a search index using atlas (https://www.mongodb.com/docs/atlas/atlas-search/tutorial/create-index/). Now I want to query it. 

I thought using this would be smort:
https://mongoosejs.com/docs/api.html#aggregate_Aggregate-search
but I can't figure out how it works. It always returns an empty array. I am also not really sure which 'path' to give since I think it would make sense to give the search index name that I created on atlas. More examples would be nice. I don't even know the possible parameters I can use.

**Comment by ViniciusMarchi** [comment]
Do you have some code example that always returns an empty array?

**Reply by vothvovo** [reply]
> Do you have some code example that always returns an empty array?

```ts
const res = await workflow.aggregate().search({
      text: {
        query: args,
        // index: "index3",
        path: "Workflow",
      },
    });
    console.log(res);
```

**Reply by vothvovo** [reply]
I can search the Table "Workflow" in my db, but I would just like to specify the index. Here is how prisma does it:
```ts
const result = await ctx.prisma.$runCommandRaw({
        aggregate: "Workflow",
        pipeline: [
          {
            $search: {
              index: "index3",
            },
          },
        ],
        cursor: {},
      });
```

**Comment by vothvovo** [comment, marked as answer]
I got it:
```ts
const workflowData: Array<WorkflowType> = await workflow
      .aggregate()
      .search({
        // queryString: { defaultPath: "title", query: args },
        text: {
          query: args,
          path: {
            wildcard: "*",
          },
        },
        index: "index3",
      });
```

I was too stupid to understand how aggregations work.

**Reply by RayyanNafees** [reply]
doesnt work with mongoose atlas search

**Reply by heijmerikx** [reply]
@RayyanNafees you use Mongoose using Atlas search, only the answer given back then might be outdated.

ie.

```javascript
const ads = await Ad.aggregate([
  {
    $search: {
      index: "dev_ads",
      text: {
        query: searchParams.get("q"),
        path: {
          wildcard: "*",
        },
      },
    },
  },
]);
```
